### ISSUE TITLE
rules-of-hooks: hooks inside try blocks are not flagged (diverges from eslint-plugin-react-hooks)

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25631]
## Summary

`react-hooks/rules-of-hooks` does not flag a hook call that is the first operation in a `try` block. This carve-out was introduced deliberately in 1.70 (#23123 → #23221, "align try/catch semantics"), but it diverges from every released version of eslint-plugin-react-hooks (4.x through current 6.1.1), which flags **all** hooks inside `try`/`catch` as conditional. The consequence during ESLint→oxlint migrations is silent under-detection: a legitimately needed `eslint-disable-next-line react-hooks/rules-of-hooks` directive is reported as unused by `--report-unused-disable-directives`, prompting users to strip a suppression the reference plugin still requires.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0 (behavior changed in 1.70 via #23221).

## Reproduction

`repro.tsx`:

```tsx
import {useTooltipContext} from './ctx'

function TestChild() {
	let captured = null
	try {
		// eslint-disable-next-line react-hooks/rules-of-hooks
		captured = useTooltipContext()
		return null
	} catch (error) {
		return null
	}
}
```

`.oxlintrc.json`:

```json
{"plugins": ["react"], "categories": {"correctness": "off"}, "rules": {"react-hooks/rules-of-hooks": ["error"]}}
```

```
oxlint -c .oxlintrc.json --report-unused-disable-directives repro.tsx
```

## Expected

With the directive removed, eslint-plugin-react-hooks flags the hook (`6:14 error React Hook "useTooltipContext" is called conditionally`) under both ESLint 7.23.0 + react-hooks 4.2.0 and ESLint 9.39.4 + react-hooks 6.1.1. With the directive present, both suppress cleanly — exit 0, no unused-directive report. The directive is meaningful and must not be reported unused.

## Actual

oxlint 1.78.0 emits no rules-of-hooks finding for the file (with or without the directive) and reports:

```
repro.tsx:6:3: warning: Unused eslint-disable directive (no problems were reported).
```

Controls confirming the rule is otherwise active under this config: the same hook inside `if (props.x)` is flagged, and a hook that is the *second* statement of the try block is flagged. Only the first-operation-in-try shape is exempted — exactly the #23221 carve-out.

The carve-out was intentional, but please consider reverting it or gating it behind an option: React's own plugin, including current 6.1.1, has no such exemption, so oxlint both accepts code the reference implementation rejects and actively tells users to delete still-needed suppressions.

## Cross-checked

| Engine | Result |
|---|---|
| ESLint 7.23.0 + eslint-plugin-react-hooks 4.2.0 + @typescript-eslint/parser 6.21.0 | flags hook at 6:14; directive suppresses cleanly, not reported unused |
| ESLint 9.39.4 + eslint-plugin-react-hooks 6.1.1 + typescript-eslint 8.62.1 | flags hook at 6:14; directive suppresses cleanly, not reported unused |
| oxlint 1.78.0 | no finding; directive reported unused at 6:3 |
